sklearn/utils/estimator_checks.py

```diff
@@ -2939,6 +2944,65 @@ def check_classifiers_regression_target(name, estimator_orig,
             e.fit(X, y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig,
+                                 strict_mode=True):
+    rng = np.random.RandomState(42)
+    estimator = clone(estimator_orig)
+    estimator_sp = clone(estimator_orig)
+    if is_regressor(estimator_orig):
+        n_samples = 10
+        n_features = 2
+        X = sparse.random(n_samples, n_features, density=0.8,
+                          random_state=rng).A.astype(np.float16)
+        X = X.astype(np.float)
+        # Less significative numbers for less numerical pbs
+        X = X[~np.all(X == 0, axis=1)]  # we remove null rows
+        assert len(np.where(X == 0)[0]) > 0  # we should have sparsity
+        assert X.shape[0] > 5  # we mustn't have very few samples
+        size_X = X.shape[0]
+        y = X.dot(np.array([0.1, -0.2])) + 0.01 * rng.randn(size_X)
+        y = y.astype(np.float16).astype(np.float)
+        # we check that limiting the precision keeps some noise
+        assert np.all(X.dot(np.array([0.1, -0.2])) != y)
+    else:
+        tags = estimator_orig._get_tags()
+        centers = 2 if tags["binary_only"] else None
+        X, y = make_blobs(random_state=rng, cluster_std=0.5, centers=centers)
+    # for put some points to zero to have a little bit of sparsity
```

Comment:
```suggestion
    # set some points to zero to have a little bit of sparsity
```

Reply:
But I don't see what this refers to

Reply:
you're right, I think it's a remainder from some previous code that I removed
I removed the comment